Replace Travis CI with GitHub Actions

thtg88 opened the pull request:
Hi there 👋
I was wondering if I could add a GitHub Actions workflow to run tests, so that I can later provide PHP 8 support for this package. Would this be of use? :)

thtg88 committed:
Add GitHub Actions CI tests workflow

thtg88 committed:
Removed services from GitHub Actions CI tests workflow

thtg88 committed:
Remove PHP 8.0 from GitHub Actions CI tests workflow

thtg88 committed:
Add PHP 7.1 and 7.2 from GitHub Actions CI tests workflow

tautvydasr commented on .github/workflows/tests.yml:
```diff
@@ -0,0 +1,40 @@
+
```
Nitpick: could we remove empty line?

thtg88 replied:
Removed ✅

tautvydasr commented on .github/workflows/tests.yml:
```diff
@@ -0,0 +1,40 @@
+
+name: tests
+
+on:
```
I am not an expert with GitHub actions but the current setup will run tests on every branch? I think it should run only on `master`:

```yml
on:
  push:
    branches: [master]
  pull_request:
    branches: [master]
```

thtg88 replied:
Updated ✅

tautvydasr commented on .github/workflows/tests.yml:
```diff
@@ -0,0 +1,40 @@
+          tools: composer:v2
+          coverage: none
+
+      - name: Install dependencies
```
Could we include composer validation step before installing dependencies?

```yml
    - name: Validate composer.json and composer.lock
      run: composer validate
```

thtg88 replied:
I've added that step in :)

tautvydasr reviewed (changes requested):
Hey @thtg88, sure contributions are very much appreciated :+1: 

Currently, the repository is using travis to run tests so to switch to GitHub actions we should do the following:

1. Remove existing `.travis.yml` file
2. Add export ignore to `.gitattributes` for GitHub workflow config files
3. Update existing `Build Status` badge in the [README.md](https://github.com/tautvydasr/blade-class-names/blame/master/README.md#L3)

thtg88 commented:
Ah my bad! Sorry I had completely missed the repo had Travis CI integrated. I'm happy to address your feedback if you are happy with the migration to GitHub Actions, considering Travis is shutting down :)

<img width="1100" alt="Screenshot 2021-02-18 at 16 30 31" src="https://user-images.githubusercontent.com/16927402/108388526-a47da500-7206-11eb-8abe-274c14a9b9db.png">

thtg88 committed:
GitHub Actions CI workflow improvements

- Remove whitespace in `tests.yml`
- Run CI only on master branch
- Validate composer dependencies before installing

thtg88 committed:
Updated Build badge in README

thtg88 commented:
I've added the badge to the README, but I believe it will only be displayed once the PR has been merged to `master` as I'm not an owner of the repo :)

thtg88 committed:
Added .github folder to gitattributes `export-ignore`

thtg88 committed:
Remove .travis.yml

thtg88 committed:
Remove .travis.yml from .gitattributes

thtg88 commented:
I believe the feedback has been addressed - let me know what you think :)

tautvydasr reviewed (approved):
Thanks for addressing the feedback :+1:
